src/documents/Contexts.md

```diff
@@ -61,15 +52,106 @@ at a particular component location in the tree are the same, as described in the
 ## Where context names are looked for ##
 
 Context names listed in a configuration are searched for at instantiation time, sequentially,
-in three kinds of "scopes" or "environments".
+in two kinds of "scopes" or "environments".
 
 * Most immediately, context names are searched for in the tree of currently instantiating
 components, searching __upwards__ through the tree from the parent of the component where the context
 name is held.
-* Following this, a search is made in the *dynamic environment*, which is notionally a separate tree
-of components bound to the currently executing stack frame (this facility is deprecated and will be
-removed in a future version of the framework)
-* Finally, context names are sought in the *static environment*, which is a tree of components
-attached globally to the JS VM (or context within it) as a whole.
+* Following this, a search is made in the *resolver root*, which is notionally a separate tree
+of components which holds all components which has been instantiated with the grade `fluid.resolveRoot` anywhere in the
+current component tree.
 
 ![Diagram showing coloured squared representing components in a component tree](images/IoC-scope.svg)
+
+## Global components: `fluid.resolveRoot` and `fluid.resolveRootSingle`
+
+All Infusion components are stored in a single-rooted **_component tree_** - this stores every component which
+has been instantiated in a particular **_Infusion context_**. This context is global to the JavaScript context 
+which has loaded Infusion - for example, a frame within a browser, or a node.js process. Even when you 
+instantiate apparently "freely-floating" components with a line such as
+
+```
+var that = fluid.myComponent();
+```
+
+these components are also assigned an address in the global tree - they are stored as immediate children
+of the global component root with a uniquely chosen name. Naturally the default visibility rules for
+these components in the component tree have to work differently for these components than they do for
+nested components described by the rules above. If every such component was in scope for every other, there 
+would be "global chaos", to coin a phrase. So all such "free components" are hidden from each other
+through a special exception in the scoping rules.
+
+However, there exist several cases when genuinely global visibility is required for some components - often
+this is accompanied by what is termed a [Singleton Design Pattern](https://en.wikipedia.org/wiki/Singleton_pattern) 
+in prevalent "Design Patterns" dialogue. As commonly described, this pattern consists of two elements - 
+
+1. Arranging for global visibility of a component (object)
+2. Arrange for there to be a single instance of a component
+
+The Infusion framework supplies two special grades, which allow the designer to arrange for both of these
+points to be controlled separately. For a start, any component which is supplied the grade `fluid.resolveRoot`,
```

Comment:
It's unclear why this grade is called <code>fluid.resolveRoot</code>. Based on the documentation, wouldn't it be more descriptive to call it something like <code>fluid.globallyVisible</code>? And below, instead of <code>fluid.resolveRootSingle</code> just <code>fluid.singleton</code>? There may be a reason for the current naming, but it isn't clear from the documentation at the moment.